pkg/cmd/template/cmd_assert_validate_test.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package template_test
+
+import (
+	cmdtpl "github.com/vmware-tanzu/carvel-ytt/pkg/cmd/template"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/files"
+
+	"testing"
+)
+
+func TestAssertValidateOnDataValues(t *testing.T) {
```

Comment:
Subtests can
- Make tests easier to read by reading the top level test and the sub-test name as one sentence.
-  Separate and group success and failure tests, and say why they succeed or fail in the name.
An idea I am not married to is something like:
```
func TestAssertValidateOnDataValuesSucceeds(t *testing.T) {
  t.Run("when all assertions pass", func(t *testing.T) {...}
}
func TestAssertValidateOnDataValuesFails(t *testing.T) {
  t.Run("when assertions fail", func(t *testing.T) {...}
  t.Run("when assertions fail on the document", func(t *testing.T) {...}
}
```

The content of the tests look great to me.

Reply:
Did some reformatting 👍